packages/core/src/Site/index.ts

```diff
@@ -101,9 +121,85 @@ const ABOUT_MARKDOWN_DEFAULT = '# About\n'
 const MARKBIND_WEBSITE_URL = 'https://markbind.org/';
 const MARKBIND_LINK_HTML = `<a href='${MARKBIND_WEBSITE_URL}'>MarkBind ${MARKBIND_VERSION}</a>`;
 
-class Site {
-  constructor(rootPath, outputPath, onePagePath, forceReload = false,
-              siteConfigPath = SITE_CONFIG_NAME, dev, backgroundBuildMode, postBackgroundBuildFunc) {
+/*
+ * A page configuration object.
+ */
+type PageCreationConfig = {
```

Comment:
Not sure about moving the types to a file, because they largely fall into 3 different purposes (page creation, page generation, deploy options) which are all quite different so extracting them to the same file may be a bit weird unless we're going for a `types.ts` kind of file for all types. We could use multiple files for the types with 1-2 types per file, which would reduce the length of Site/index.ts, but I feel that it doesn't add much in terms of understandable-ness especially since most of these types are only used once or a few times in Site/index.ts. But then again I don't think it hurts either, besides having the types further from where they are defined 🤔  Wdyt?

Reply:
Ahh okay. I think it is fine to leave it then since the types are only used in this file.